tests/test_run_reference_vmm.py

```diff
@@ -423,3 +444,80 @@ def test_reference_vmm_mem(kernel):
         expect_mem(vmm_process, expected_mem_mib)
 
         shutdown(vmm_process)
+
+
+@pytest.mark.parametrize("kernel,disk", UBUNTU_KERNEL_DISK_PAIRS)
+def test_reference_vmm_with_net_device(kernel , disk):
+    """Start the reference VMM and verify the tap device."""
+
+    prompt = "root@ubuntu-rust-vmm:~#"
+    # These are defults values for net devices.
+    tap_device = "tap0"
+    guest_ip = "172.16.0.2"
+    host_ip = "172.16.0.1"
+    mask = 24
+    guest_ip_with_mask = f"{guest_ip}/{mask}"
+
+    setup_host()
+
+    try:
+        vmm_process, temp_file_path = start_vmm_process(kernel, disk_path=disk ,tap_device=tap_device)
+
+        setup_guest(vmm_process , prompt)
+        
+        # ping the host from guest to verify that the network interface works
+        ping_cmd = f"ping -c 2 {host_ip}"
+        output = run_cmd_inside_vm(ping_cmd.encode(),vmm_process,prompt.encode(),timeout=10)
+        
+        output = b''.join(output).decode()
+        result = output.find("2 received")
+        assert result != -1
+    
+    except:
+        raise
+    
+    finally:
+        shutdown(vmm_process)
+        if temp_file_path:
+            os.remove(temp_file_path)
+
+        # we will always clear the tap device no matter if the
+        # test passes or fails.
+        clean_tap()
+    
+
+def setup_guest(vmm_process , prompt , guest_ip_with_mask="172.16.0.2/24" , host_ip="172.16.0.1"):
+    """Configure the guest vm with the tap device"""
+    
+    # To clear the inital stdout which was printed while 
+    # booting the vmm.
+    expect_string(vmm_process,prompt)
+    
+    add_addr_cmd = f"ip addr add {guest_ip_with_mask} dev eth0"
```

Comment:
The name of the interface should also be sent as a parameter because it depends on the image with which you test.